### PR TITLE
Add MEP for dedicated docker driver

### PR DESCRIPTION
**Pull request by medyagh**

### CODE REVIEW

File: enhancements/proposed/20200921-dedicated-network-kic/dedicated-network-kic.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+container_index is calculated from this formula:
+
+default is 1 for single node (and for multinde is the order the machine created based on it is name, for example the index for minikube-m02 will be 2)
+
+
+
+### Alternative design :
+
+once network is found or created, minikube can look up list of all the container IPs in the network, and try to 
+
```
Comment (tstromberg) [inline enhancements/proposed/20200921-dedicated-network-kic/dedicated-network-kic.md:28]: Can you clarify why this approach isn't ideal?

File: enhancements/proposed/20200921-dedicated-network-kic/dedicated-network-kic.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Static IP for minikube containers KIC Drivers (docker/podman)
+
+design by @medyagh
+
```
Comment (tstromberg) [inline enhancements/proposed/20200921-dedicated-network-kic/dedicated-network-kic.md:4]: Can you include the rest of the standard MEP template text?

File: enhancements/proposed/20200921-dedicated-network-kic/dedicated-network-kic.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+## Prototype [PR 9294](https://github.com/kubernetes/minikube/pull/9294):
+
+Each minikube cluster will get it is own network, instead of using docker’s owned default network which is unmodifiable by the user.
```
Comment (tstromberg) [inline enhancements/proposed/20200921-dedicated-network-kic/dedicated-network-kic.md:7]: s/owned/own/g

File: enhancements/proposed/20200921-dedicated-network-kic/dedicated-network-kic.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+on “minikube start”, minikube will try to create a docker network at a default subnet
```
Comment (tstromberg) [inline enhancements/proposed/20200921-dedicated-network-kic/dedicated-network-kic.md:9]: Can you describe how the default subnet was chosen, and what occurs if that default subnet happens to be the one the host is using?

File: enhancements/proposed/20200921-dedicated-network-kic/dedicated-network-kic.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+
+### Alternative design :
+
+once network is found or created, minikube can look up list of all the container IPs in the network, and try to 
+
+
+### Considerations:
+
+if can’t create network, should fall back to docker’s default bridge IP
```
Comment (tstromberg) [inline enhancements/proposed/20200921-dedicated-network-kic/dedicated-network-kic.md:32]: Move this to the standard Goals section of the MEP template.

File: enhancements/proposed/20200921-dedicated-network-kic/dedicated-network-kic.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+The existing running clusters should not break by using newer version of minikube
+
+
+### Advantages:
+
+since we own the created network we can assign and calculate the IP.
```
Comment (tstromberg) [inline enhancements/proposed/20200921-dedicated-network-kic/dedicated-network-kic.md:38]: Could you clarify thiss statement?

**Review by tstromberg — CHANGES_REQUESTED** [review]

**Review by priyawadhwa — APPROVED** [review]

**Pull request merged**